applications/RANSApplication/custom_strategies/algebraic_flux_corrected_scalar_steady_scheme.h

```diff
@@ -0,0 +1,567 @@
+#pragma omp parallel for
+            for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
+            {
+                const ModelPart::ConditionType& r_condition =
+                    *(rModelPart.ConditionsBegin() + i_cond);
+                if (r_condition.Is(PERIODIC))
+                {
+                    // this only supports 2 noded periodic conditions
+                    KRATOS_ERROR_IF(r_condition.GetGeometry().PointsNumber() != 2)
+                        << this->Info() << " only supports two noded periodic conditions. Found "
+                        << r_condition.Info() << " with "
+                        << r_condition.GetGeometry().PointsNumber() << " nodes.\n";
+
+                    const ModelPart::NodeType& r_node_0 = r_condition.GetGeometry()[0];
+                    const std::size_t r_node_0_pair_id =
+                        r_node_0.FastGetSolutionStepValue(mrPeriodicIdVar);
+
+                    const ModelPart::NodeType& r_node_1 = r_condition.GetGeometry()[1];
+                    const std::size_t r_node_1_pair_id =
+                        r_node_1.FastGetSolutionStepValue(mrPeriodicIdVar);
+
+                    KRATOS_ERROR_IF(r_node_0_pair_id != r_node_1.Id())
+                        << "Periodic condition pair id mismatch in "
+                        << mrPeriodicIdVar.Name() << ". [ " << r_node_0_pair_id
+                        << " != " << r_node_1.Id() << " ].\n";
+
+                    KRATOS_ERROR_IF(r_node_1_pair_id != r_node_0.Id())
+                        << "Periodic condition pair id mismatch in "
+                        << mrPeriodicIdVar.Name() << ". [ " << r_node_1_pair_id
+                        << " != " << r_node_0.Id() << " ].\n";
+                }
+            }
+        }
+
+        // Allocate auxiliary memory.
+        const auto num_threads = OpenMPUtils::GetNumThreads();
+        mAntiDiffusiveFlux.resize(num_threads);
+        mAntiDiffusiveFluxCoefficients.resize(num_threads);
+        mValues.resize(num_threads);
+        mAuxMatrix.resize(num_threads);
+
+        KRATOS_CATCH("");
+    }
+
+    void InitializeNonLinIteration(ModelPart& rModelPart,
+                                   TSystemMatrixType& A,
+                                   TSystemVectorType& Dx,
+                                   TSystemVectorType& b) override
+    {
+        KRATOS_TRY
+
+        ModelPart::NodesContainerType& r_nodes = rModelPart.Nodes();
+
+        VariableUtils variable_utilities;
+        variable_utilities.SetHistoricalVariableToZero(
+            AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX, r_nodes);
+        variable_utilities.SetHistoricalVariableToZero(
+            AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX, r_nodes);
+        variable_utilities.SetHistoricalVariableToZero(
+            AFC_POSITIVE_ANTI_DIFFUSIVE_FLUX_LIMIT, r_nodes);
+        variable_utilities.SetHistoricalVariableToZero(
+            AFC_NEGATIVE_ANTI_DIFFUSIVE_FLUX_LIMIT, r_nodes);
```

Comment:
In this case I think it makes sense to manually do the loop in here to avoid looping the nodes 4 times.

Reply:
done :+1: